apps/admin_app/lib/admin_app_web/controllers/product_controller.ex

```diff
@@ -115,8 +115,16 @@ defmodule AdminAppWeb.ProductController do
           "theme_id" => params["theme_id"]
         })
 
-      {:ok, product} = Repo.update(changeset)
-      redirect(conn, to: product_path(conn, :index))
+      case changeset.valid? do
```

Comment:
Can you put this also in `with` pipeline? You can handle the false case in the else part.

Reply:
done